lib/auth/grpcserver.go

```diff
@@ -1886,6 +1886,84 @@ func (g *GRPCServer) IsMFARequired(ctx context.Context, req *proto.IsMFARequired
 	return resp, nil
 }
 
+// GetTrustedCluster retrieves a Trusted Cluster by name.
+func (g *GRPCServer) GetTrustedCluster(ctx context.Context, req *types.ResourceRequest) (*types.TrustedClusterV2, error) {
+	auth, err := g.authenticate(ctx)
+	if err != nil {
+		return nil, trail.ToGRPC(err)
+	}
+	tc, err := auth.ServerWithRoles.GetTrustedCluster(ctx, req.Name)
+	if err != nil {
+		return nil, trail.ToGRPC(err)
+	}
+	trustedClusterV2, ok := tc.(*types.TrustedClusterV2)
+	if !ok {
+		return nil, trail.ToGRPC(trace.Errorf("encountered unexpected Trusted Cluster type %T", tc))
```

Comment:
```suggestion
		return nil, trail.ToGRPC(trace.BadParameter("encountered unexpected Trusted Cluster type %T", tc))
```

Reply:
Doesn't `Errorf` make more sense here? It's unrelated to parameters.

Reply:
The similar cases (`unexpected type`) are using `BadParameter` hence I wanted to point it out. Feel free to leave it as-is though - not that important.

Reply:
Gotcha, I changed all of them to have same error message and use Errorf, thanks!